File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numset=set(nums)
        starting=set()
        for i in nums:
            if i-1 in numset:
                continue
            else:
                starting.add(i)
        for i in starting:
            numset.remove(i)
        max_count=0
        for i in starting:
            nextn=i+1
            count=1
            while (nextn in numset):
                numset.remove(nextn)
                count+=1
                nextn+=1
            max_count=max(count,max_count)
        return max_count

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> num_set=new HashSet<Integer>();
        for (int num: nums ){
            num_set.add(num);
        }
        int max_count=0;
        for (int num: num_set){
            if (!num_set.contains(num-1)){
                int count=1;
                int curr=num;
                while(num_set.contains(num+1)){
                    num+=1;
                    count+=1;
                }
                max_count=Math.max(max_count, count);
            }
        }
        return max_count;
    }
}